### PR TITLE
souffle: 2.0.2 -> 2.1

### PR DESCRIPTION
**Pull request by langston-barrett**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/souffle/default.nix [hunk 1]
```diff
@@ -10,31 +10,30 @@ let
...
     owner  = "souffle-lang";
     repo   = "souffle";
     rev    = version;
-    sha256 = "1fa6yssgndrln8qbbw2j7j199glxp63irfrz1c2y424rq82mm2r5";
+    sha256 = "11x3v78kciz8j8p1j0fppzcyl2lbm6ib4svj6a9cwi836p9h3fma";
   };
 
-  nativeBuildInputs = [ autoreconfHook bison flex mcpp doxygen graphviz makeWrapper perl ];
-  buildInputs = [ ncurses zlib sqlite libffi ];
+  nativeBuildInputs = [ bison cmake flex lsb-release git mcpp doxygen graphviz makeWrapper perl ];
```
[the remark this answers is not in the document]
Comment (langston-barrett) [reply pkgs/development/compilers/souffle/default.nix]: I tried to work around these dependencies, but with the changes I made the build no longer works, nor does it output any useful error messages:
```
these derivations will be built:
  /nix/store/l448hi41b2nwbs302fy3j1kd9gg7ir3b-souffle-2.1.drv
building '/nix/store/l448hi41b2nwbs302fy3j1kd9gg7ir3b-souffle-2.1.drv'...
builder for '/nix/store/l448hi41b2nwbs302fy3j1kd9gg7ir3b-souffle-2.1.drv' failed with exit code 1
error: build of '/nix/store/l448hi41b2nwbs302fy3j1kd9gg7ir3b-souffle-2.1.drv' failed
```

File: pkgs/development/compilers/souffle/default.nix [hunk 2]
```diff
@@ -1,40 +1,43 @@
...
   # since generated compiler code uses them. TODO: maybe write a g++ wrapper
   # that adds these so we can keep the propagated inputs clean?
   propagatedBuildInputs = [ ncurses zlib sqlite libffi ];
 
-  # see 565a8e73e80a1bedbb6cc037209c39d631fc393f and parent commits upstream for
-  # Wno-error fixes
-  patchPhase = ''
-    substituteInPlace ./src/Makefile.am \
-      --replace '-Werror' '-Werror -Wno-error=deprecated -Wno-error=other'
+  patches = [ ./bash-completion-dir.patch ./lsb-release.patch ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/souffle/default.nix:34]:
```suggestion
  patches = [
    ./bash-completion-dir.patch
    ./lsb-release.patch
  ];
```

File: pkgs/development/compilers/souffle/default.nix [hunk 3]
```diff
@@ -1,40 +1,43 @@
...
+, makeWrapper, writeShellScript
 }:
 
 
 let
   toolsPath = lib.makeBinPath [ mcpp ];
+  # The build system of Soufflé 2.1 requires git on the $PATH, but will without
+  # it as long as it exists with a non-zero code. This should be unnecessary in
+  # the next release, see SOUFFLE_GIT in CMakeLists.txt.
+  git = writeShellScript "git" "exit 1";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/souffle/default.nix:13]: Can we patch this instead?

Comment (langston-barrett) [reply pkgs/development/compilers/souffle/default.nix:13]: Sure, happy to. Why is patching preferred, out of curiosity?

File: pkgs/development/compilers/souffle/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/compilers/souffle/default.nix]: You did not copy the interesting parts of the log. That only shows that the build failed but now why.

File: pkgs/development/compilers/souffle/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/compilers/souffle/default.nix:13]: Creating fake binaries is more a hack. Patching the right version number in the build files is nicer and makes sure it is set correct.

**Comment by langston-barrett** [comment]
Closing in favor of #164103

**Pull request closed without merging**